dali/operators/reader/coco_reader_op.h

```diff
@@ -104,8 +114,16 @@ class COCOReader : public DataReader<CPUBackend, ImageLabelWrapper> {
         coords.size() * sizeof(float));
     }
 
+    if (pixelwise_masks_) {
+      auto &masks_output = ws.Output<CPUBackend>(3);
+      masks_output.Resize({1, widths_[image_id], heights_[image_id]});
```

Comment:
Well the thing is, run-length encoding implies a width-first scan, so it seemed natural to output CWH or WHC directly.

Of course I could just transpose afterwards. But in my pipeline I just use an Operator for that during data augmentation, and that seems to do the trick..

I'll add the SetLayout.

Reply:
If RLEs are scanned first horizontally, then vertically, then the storage order should be CHW or HWC. The layouts and shapes have the _outer_ dimension first.

Reply:
Yeah sorry, my mistake.
pycocotools actually use column-major order : https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/mask.py#L51
so I meant height-first scan.

Typically I use
https://numpy.org/doc/stable/reference/generated/numpy.asfortranarray.html
when creating my annotations for that very reason.

Reply:
So, what's the plan? I don't think it's good to produce a mask that has a layout incompatible with the image (which is row-major) unless an overwhelming majority of the models expect the mask in that order. Otherwise, we should go with row-major or perhaps configurable layout (swapping the axes is relatively straightforward - we just need to swap w, h and the coordinates of the points passed to rleFrPoly).

Reply:
>  I don't think it's good to produce a mask that has a layout incompatible with the image (which is row-major) unless an overwhelming majority of the models expect the mask in that order.

Can't argue with that, I'd like to output HWC. I'm not sure what's the best way though, aside from the obvious transpose and a swap buffer after the final decoding.

> we just need to swap w, h and the coordinates of the points passed to rleFrPoly

The problem is that some annotations are already in column-major RLE (the ones we pass through rleFrString).
Of course since I decode manually at the end of the function I could just write the row-major tensor column-by-column, but that's probably going to be a very inefficient memory access scheme, and the decoding is very clearly the bottleneck.

I can't figure out a way to convert from column-major RLE to row-major RLE.

What do you suggest @mzient ?

Reply:
1. Are we 100% sure that there are situations where the masks are indeed column major and the image is not?
2. Do we need to have a transposed RLE or just transposed mask? If it's the latter, then it's quite easy to render the mask in the other direction.

Reply:
1. Yeah as soon as you use pycocotools, you'll get column-major masks with RLE. And since it's the "official" tool...
2. Yup got it !